### PR TITLE
Fix 422 clip bugs report by ffmpeg vvdec integrate

### PR DESCRIPTION
**Pull request by chenm001**
* minQT size mistake
  * DMVR
  8b422_A_Sony_3
  8b422_B_Sony_3
  8b422_C_Sony_3
  8b422_D_Sony_3
  8b422_E_Sony_3
  8b422_F_Sony_3
  10b422_A_Sony_3
  10b422_B_Sony_3
  10b422_C_Sony_3
  10b422_D_Sony_3
  10b422_E_Sony_3
  10b422_F_Sony_3

### CODE REVIEW

**Comment by adamjw24** [comment]
So this one will have to wait until after the holidays to be discussed internally. I very much appreciate your work and I'm sure we'll integrate this then, though.

The problem with 422 is that it is the part of the main10-444, for which there is more to do than fix just chroma scaling. Proper support would also require ACT and palette-mode support, which are still pending in VVdeC.

**Comment by chenm001** [comment]
Implement full sets of Main10-444 are large project workload, there have many new features, such as palette-mode.
In my view, the ffmpeg patches report vvdec pass 68% (163/238), it sounds not good coverage enough, so I try to fix as many bugs as possible, it became 207/238 now, it maybe good start point for video product.

**Comment by adamjw24** [comment]
I totally agree.

Still, I'd like to defer the integration until we discuss it internally. Should be in the first week of 2021.

**Comment by chenm001** [comment]
Sure, Merry Christmas.

File: source/Lib/CommonLib/CodingStructure.cpp [hunk 1]
```diff
@@ -525,9 +525,10 @@ void CodingStructure::fillIBCbuffer( CodingUnit &cu, int lineIdx )
         continue;
 
       const unsigned int lcuWidth = sps->getMaxCUWidth();
-      const int shiftSample = ::getComponentScaleX(area.compID, cu.chromaFormat);
-      const int ctuSizeLog2 = getLog2(lcuWidth) - shiftSample;
-      const int pux = area.x & ((m_IBCBufferWidth >> shiftSample) - 1);
+      const int shiftSampleHor = ::getComponentScaleX(area.compID, cu.chromaFormat);
+      const int shiftSampleVer = ::getComponentScaleY(area.compID, cu.chromaFormat);
+      const int ctuSizeLog2 = getLog2(lcuWidth) - shiftSampleVer;
```
Comment (adamjw24) [inline source/Lib/CommonLib/CodingStructure.cpp:530]: Could you rename it to ctuSizeVerLog2?

File: source/Lib/CommonLib/InterPrediction.cpp [hunk 2]
```diff
@@ -2093,8 +2097,9 @@ void InterPrediction::xFillIBCBuffer(CodingUnit &cu)
 void InterPrediction::xIntraBlockCopy( PredictionUnit &pu, PelUnitBuf &predBuf, const ComponentID compID )
 {
   const unsigned int lcuWidth = pu.slice->getSPS()->getMaxCUWidth();
-  int shiftSample = ::getComponentScaleX( compID, pu.chromaFormat );
-  const int ctuSizeLog2 = getLog2(lcuWidth) - shiftSample;
+  const int shiftSampleHor = ::getComponentScaleX( compID, pu.chromaFormat );
+  const int shiftSampleVer = ::getComponentScaleY( compID, pu.chromaFormat );
+  const int ctuSizeLog2 = getLog2(lcuWidth) - shiftSampleVer;
```
Comment (adamjw24) [inline source/Lib/CommonLib/InterPrediction.cpp:2102]: Same as above, could this be ctuSizeVerLog2?

Comment (chenm001) [reply source/Lib/CommonLib/InterPrediction.cpp]: Thank your comments. I have been modified and rebase my patch.

**Pull request merged**